drivers/sensor/aosong/dht20/dht20.c

```diff
@@ -147,16 +150,15 @@ static inline int reset_sensor(const struct device *dev)
 		if (rc < 0) {
 			return rc;
 		}
-		/* Wait 10ms after reset sequence */
-		k_msleep(DHT20_RESET_SEQUENCE_WAIT_MS);
+		/* Wait 10ms after status reset sequence */
+		k_msleep(DHT20_STATUS_RESET_SEQUENCE_WAIT_MS);
 	}
 
 	return 0;
 }
 
-static int dht20_read_sample(const struct device *dev, uint32_t *t_sample, uint32_t *rh_sample)
+static int dht20_read_sample(struct dht20_data *data, uint8_t *rx_buf)
```

Comment:
Removing `rx_buf` to then receive a pointer hoping it's the same size does not look like a good idea. I suggest keeping the local buffer.
```suggestion
static int dht20_read_sample(struct dht20_data *data)
```